internal/controller/nonadminbackup_predicate.go

```diff
@@ -0,0 +1,67 @@
+// velerobackup_predicate.go
+
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	nacv1alpha1 "github.com/migtools/oadp-non-admin/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type NonAdminBackupPredicate struct {
+	Logger logr.Logger
+}
+
+func getNonAdminBackupPredicateLogger(ctx context.Context, name, namespace string) logr.Logger {
+	return log.FromContext(ctx).WithValues("NonAdminBackupPredicate", types.NamespacedName{Name: name, Namespace: namespace})
+}
+
+func (predicate NonAdminBackupPredicate) Create(ctx context.Context, evt event.CreateEvent) bool {
+
+	if _, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
+		nameSpace := evt.Object.GetNamespace()
+		name := evt.Object.GetName()
+		log := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
+		log.V(1).Info("Received Create NonAdminBackupPredicate")
+		return true
+	}
+
+	return false
+}
+
+func (predicate NonAdminBackupPredicate) Update(ctx context.Context, evt event.UpdateEvent) bool {
+	if _, ok := evt.ObjectNew.(*nacv1alpha1.NonAdminBackup); ok {
+		nameSpace := evt.ObjectNew.GetNamespace()
+		name := evt.ObjectNew.GetName()
+		log := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
+		log.V(1).Info("Received Update NonAdminBackupPredicate")
+		return true
+	}
+	return false
+}
+
+func (predicate NonAdminBackupPredicate) Delete(ctx context.Context, evt event.DeleteEvent) bool {
+	if _, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
+		nameSpace := evt.Object.GetNamespace()
+		name := evt.Object.GetName()
+		log := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
+		log.V(1).Info("Received Delete NonAdminBackupPredicate")
+		return true
+	}
+	return false
+}
+
+func (predicate NonAdminBackupPredicate) Generic(ctx context.Context, evt event.GenericEvent) bool {
```

Comment:
generic would be the default of switch case?

Reply:
Here you have more about those 4 predicates, it's interface which we need to implement:
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.2/pkg/predicate/predicate.go#L32